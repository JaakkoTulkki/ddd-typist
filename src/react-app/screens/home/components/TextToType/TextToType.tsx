import React from 'react';

interface TexToTypeProps {
    text: string;
}

export const TextToType: React.SFC<TexToTypeProps> = ({text}) =>
    <div data-test-id="text-to-type">{text}</div>;