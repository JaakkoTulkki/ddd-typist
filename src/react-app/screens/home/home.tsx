import React from 'react';
import { TextToType } from "./components/TextToType/TextToType";

export const Home = () =>
    <div>
        <TextToType />
    </div>;
