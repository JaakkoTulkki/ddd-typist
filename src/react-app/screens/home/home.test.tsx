import React from 'react';
import { mount } from 'enzyme';
import {Home} from "./home";
import { TextToType } from "./components/TextToType/TextToType";

describe('Home', () => {
    it('should render TextToType', () => {
        const home = mount(<Home />);
        expect(home.find(TextToType).length).toEqual(1);
    });
});