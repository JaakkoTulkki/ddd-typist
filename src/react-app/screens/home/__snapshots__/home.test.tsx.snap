// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <div>
        home is home
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "home is home",
      },
      "ref": null,
      "rendered": Array [
        "home is home",
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <div>
          home is home
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "home is home",
        },
        "ref": null,
        "rendered": Array [
          "home is home",
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
